### PR TITLE
Feat/architecture

### PR DESCRIPTION
**Pull request by reganlosey**
What (if any) features are you implementing?
- Created an instance of the Idea class when a card is saved.
- Implement JSON functionality.
- Build functionality to display user's card on the DOM in a grid formation.
- Build functionality so that cards will persist on page reload.
- Set the save button to default to disabled and gray when the input fields are empty.

### CODE REVIEW

File: Idea.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    this.title = title;
+    this.body = body;
+    this.star = false;
+  }
+
+saveToStorage(ideas) {
+  var stringifiedIdeas = JSON.stringify(ideas);
+  localStorage.setItem('ideas', stringifiedIdeas);
+}
+// deleteFromStorage
```
Comment (ivonne-hernandez) [inline Idea.js:15]: We can go back and delete the pseudocode before we push.

File: Idea.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+deleteFromStorage() {
```
Comment (ivonne-hernandez) [inline Idea.js:16]: We can go back and delete the pseudocode before we push.

File: Idea.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  var retrievedIdea = localStorage.getItem('idea');
+  //this specific idea's identity will need to be handled in main.js, when invoked/found this param will be passed into this function
+  localStorage.removeItem(retrievedIdea);
+  // var parsedIdea = JSON.parse(retrievedIdea);
+}
+// updateIdea (should update the idea’s starred state)
+
+
+}
+//no DOM manipulation in classes
```
Comment (ivonne-hernandez) [inline Idea.js:26]: We can go back and delete the pseudocode before we push.

File: main.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  ideas.push(savedIdea);
+  displayCards();
+  savedIdea.saveToStorage(ideas);
+  if(titleInput && bodyInput){
+    titleInput.value = "";
+    bodyInput.value = "";
+    saveButton.disabled = true;
+  }
+}
+//make another change;
```
Comment (ivonne-hernandez) [inline main.js:33]: We can delete all of this pseudocode.

File: main.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+  if (userTitle && userBody) {
+    saveButton.disabled = false;
+  } else {
+    saveButton.disabled = true;
+    return;
+  }
+}
+
+//make sure both input fields hold data before save button can be clicked:
```
Comment (ivonne-hernandez) [inline main.js:61]: We can go back and delete this.

File: main.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      <div class="box-footer-container">
+        <button class="comment-btn"><img src="./assets/comment.svg"></button>
+        <label class="comment-label">Comment</label>
+      </div>
+    </div>`;
+  }
+}
+
+function deleteIdea() {
+  //get the id of the target that the user clicked on (using a for loop).
```
Comment (ivonne-hernandez) [inline main.js:89]: We can go back and delete the pseudocode before we push.

File: main.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+
+
+
+
+
+
+
+
+//var starredIdeas = [display when user clicks on "Show Starred Ideas"];
```
Comment (ivonne-hernandez) [inline main.js:108]: We can go back and delete this.

**Review by ivonne-hernandez — APPROVED** [review]
We'll have to make sure to delete the pseudocode before we push to GH on our next iteration.

**Comment by edwardkrupicka** [comment]
Everything is looking great so far team. We need to get pushing on iteration 3 so I can eat my chili in peace.

**Pull request merged**